### DISCUSSION TITLE
Docker command Hang

### DISCUSSION THREAD

**Discussion by week2311** [moby/moby#44814] [Q&A]
Description：
My kubernetes clusters often becomes uhealth after a while，and the node is NotReady。I have find out why，because the status of container service is hang above uhealth node。It's regular occurrence，about a week.
I can restart docker serivce to resolve problem，but it's a temporary solution. Now，I read this issue and foud that i can use 'strace -p {dockerd-pid} -f' command to reslove，but i don't know why。
https://github.com/moby/moby/issues/28889。

Docker Version：
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea838
 Built:             Wed Nov 13 07:22:05 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea838
  Built:            Wed Nov 13 07:28:45 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683

Kernel Version：5.4.217-1.el7.elrepo.x86_64

**Comment by thaJeztah** [comment]
Docker 19.03 is pretty old, and both Docker 19.03.x and the containerd and runc versions you're running reached EOL (and are no longer maintained). If you have a test-environment in which you can reproduce the issue, it would be useful to verify if the issue still reproduces on a current version of docker (currently 20.10.x)

**Comment by corhere** [comment]
It's a 5.4.21x LTS kernel again, just like #44199 and #44763. If you are able to get a current version of dockerd to hang, please terminate the dockerd process with `kill -ABRT` _while it is hung_ and attach the standard error output of the terminated dockerd process to #44763. The goroutine stack dumps would be very helpful to troubleshoot this issue. Thanks!
